charts/kong/templates/_helpers.tpl

```diff
@@ -535,7 +535,22 @@ The name of the service used for the ingress controller's validation webhook
   {{- toYaml .Values.resources | nindent 4 }}
 {{- end -}}
 
+{{/*
+kong.controller2xplus returns "true" if the controller image tag major version is >= 2
+Note that this is a string, not a boolean, because templates vov
+*/}}
+{{- define "kong.controller2xplus" -}}
+{{- $version := "" -}}
+{{- if .Values.ingressController.image.effectiveSemver -}}
```

Comment:
consider using the `default` template for brevity instead of if/else

Reply:
This:
```
{{- define "kong.controller2xplus" -}}
{{- $version := default (semver .Values.ingressController.image.tag) (semver .Values.ingressController.image.effectiveSemver) -}}
{{- ge $version.Major 2 -}}
{{- end -}}
```
is cleaner, but `default` evaluates all its arguments immediately, so it tries to call `semver` on the tag even if `effectiveSemver` isn't empty, and calls it on `effectiveSemver` if it isn't set. 415b48fdd does improve on it a bit though, by defaulting strings and putting the conversion after.